dashboard/src/actions/endpointAction.js

```diff
@@ -12,6 +13,15 @@ export const fetchEndpoints = async (dispatch) => {
       type: TYPES.SET_ENDPOINTS,
       payload: data,
     });
+    const keycloak = new Keycloak({
+      url: data.openid.server,
+      realm: data.openid.realm,
+      clientId: data.openid.client,
```

Comment:
Yeah, what Webb said. Having `openid` undefined is not acceptable here, if the value of `openid` is undefined the value of the server URL would be a stringified `undefined` if we used the `?` chaining operation and that's gonna result in an error creating a new Keycloak object. So why not raise an error on the fetch of `data.openid` instead of when creating a Keycloak object.

Reply:
Got it!